.github/workflows/julia.yml

```diff
@@ -0,0 +1,57 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Julia
+on:
+  push:
+    paths:
+      - '.github/workflows/julia.yml'
+      - 'ci/docker/*_julia.dockerfile'
+      - 'julia/**'
+  pull_request:
+    paths:
+      - '.github/workflows/julia.yml'
+      - 'ci/docker/*_julia.dockerfile'
+      - 'ci/docker/**'
+      - 'julia/**'
+
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    env:
+      JULIA_NUM_THREADS: 2
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.3'
+          - '1' # automatically expands to the latest stable 1.x release of Julia
+          - 'nightly'
+        os:
+          - ubuntu-latest
```

Comment:
Is it enough to run this job only on Ubuntu?
Do we need to run this job on macOS and Windows too?

Reply:
I've added windows too; we don't expect there to be any platform differences between linux/osx, so we stopped testing that a while ago to try and be better citizens around free github action minutes. We can test windows too, but again, the julia binary works out of the box well across platforms and we're not doing anything in the Arrow.jl code that is very platform-specific at all.

Reply:
IMO this is fine for now, but if we become concerned about the number of GHA jobs/minutes, we can move some of these builds to nightly/on-demand. Like, run Windows w/Julia current release and Linux w/Julia nightly on every commit, and with those 2 jobs you get most of the coverage you care about; then move the other parts of the build matrix to nightly jobs.

Reply:
@quinnj I see.

@nealrichardson +1